src/AppInstallerRepositoryCore/Microsoft/Schema/1_4/DependenciesTable.cpp

```diff
@@ -0,0 +1,375 @@
+                    manifestVersions.end(),
+                    [&](auto& manifestVersion)
+                    {
+                        return (std::get<0>(manifestVersion) == depMinVersion.ToString());
+                    }
+                );
+
+                if (result == manifestVersions.end())
+                {
+                    VersionTable::EnsureExists(connection, depMinVersion.ToString());
+                }
+
+                idsMap[packageId] = packageRowId.value();
+                versionsMap[depMinVersion] = VersionTable::SelectIdByValue(connection, depMinVersion.ToString()).value();
+            }
+
+            ThrowOnMissingPackageNodes(missingPackageNodes);
+
+            StatementBuilder insertBuilder;
+            insertBuilder.InsertInto(s_DependenciesTable_Table_Name)
+                .Columns({ s_DependenciesTable_Manifest_Column_Name, s_DependenciesTable_MinVersion_Column_Name, s_DependenciesTable_PackageId_Column_Name })
+                .Values(Unbound, Unbound, Unbound);
+            SQLite::Statement insert = insertBuilder.Prepare(connection);
+
+            for (const auto& dep : dependencies)
+            {
+                insert.Reset();
+                insert.Bind(1, manifestRowId);
+                insert.Bind(2, versionsMap[dep.MinVersion.value().ToString()]);
+                insert.Bind(3, idsMap[dep.Id]);
+
+                insert.Execute(true);
+            }
+        }
+
+        std::optional<std::pair<SQLite::rowid_t, Utility::NormalizedString>> GetPackageRowIdAndVersion(SQLite::Connection& connection, const Manifest::Dependency& dependency)
+        {
+            std::optional<std::pair<SQLite::rowid_t, Utility::NormalizedString>> row;
+
+            auto packageId = IdTable::SelectIdByValue(connection, dependency.Id);
+
+            if (packageId.has_value())
+            {
+                row = std::make_pair(packageId.value(), dependency.MinVersion.value().ToString());
+            }
+
+            return row;
+        }
+    }
+
+    std::string_view DependenciesTable::TableName()
+    {
+        return s_DependenciesTable_Table_Name;
+    }
+
+    void DependenciesTable::Create(SQLite::Connection& connection)
+    {
+        using namespace SQLite::Builder;
+        SQLite::Savepoint savepoint = SQLite::Savepoint::Create(connection, "createDependencyTable_v1_4");
+
+        StatementBuilder createTableBuilder;
+        createTableBuilder.CreateTable(TableName()).BeginColumns();
+        createTableBuilder.Column(IntegerPrimaryKey());
+
+        std::vector<DependenciesTableColumnInfo> dependenciesColumns =
+        {
+            { s_DependenciesTable_Manifest_Column_Name },
+            { s_DependenciesTable_MinVersion_Column_Name },
+            { s_DependenciesTable_PackageId_Column_Name }
+        };
+
+        for (const DependenciesTableColumnInfo& value : dependenciesColumns)
+        {
+            createTableBuilder.Column(ColumnBuilder(value.Name, Type::RowId).NotNull());
```

Comment:
>NotNull [](http://example.com/codeflow?start=77&length=7)

MinVersion is optional.  This could be modeled by an empty string for the version, but it isn't clear that you are doing so.